### PR TITLE
fix submit function via url

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Submitting a function in which the function package is a url is broker right now.  If a user is submitting a function with a url package, he or she will get the following exception:

```
Jerrys-MBP:incubator-pulsar jerrypeng$ ./bin/pulsar-admin functions create --function-config-file ~/function.yaml 
User class must be in class path

Reason: User class must be in class path
```

The reason is because we don't actually download the function package from the URL and load it

### Modifications

Download the JAR from the function package URL.

I am not sure why in the code we only download the first 10 bytes from the URL and then delete the file.  Was this just to check if the URL is valid? 

Maybe @rdhabalia can chime in since he wrote most of this code originally.

Though not sure we should download the function package/JAR so that we can do some validation.  The JAR can be very large.

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java [hunk 1]
```diff
@@ -385,8 +385,7 @@ public void testE2EPulsarFunction() throws Exception {
 
     }
 
-    @Test(timeOut = 20000)
-    public void testPulsarSinkStats() throws Exception {
+    public void testPulsarSinkStats(String jarFilePathUrl) throws Exception {
```
Comment (sijie) [inline pulsar-broker/src/test/java/org/apache/pulsar/io/PulsarFunctionE2ETest.java]: private

**Comment by merlimat** [comment]
@jerrypeng 

There seems to be genuine test failure:

```
java.lang.NullPointerException
	at org.apache.pulsar.io.PulsarFunctionE2ETest.testPulsarSinkStats(PulsarFunctionE2ETest.java:424)
	at org.apache.pulsar.io.PulsarFunctionE2ETest.testPulsarSinkStatsWithUrl(PulsarFunctionE2ETest.java:595)
```

**Comment by jerrypeng** [comment]
@merlimat this just means the prometheus stats did not have the function stats in them.  Its a timing issue because I am trying to download the jar from a url.  I am working on a fix for this

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
04de22e

**Pull request merged**